mozilla-services/consvc-shepherd
# [SHEP-37] Fix the way the sync script updates existing Deals

## References

[SHEP-37](https://mozilla-hub.atlassian.net/browse/SHEP-37)

## Problem Statement

We're seeing [errors in deployed environments](https://mozilla.sentry.io/issues/5805480794/?project=4505206579462144&query=IntegrityError&referrer=issue-stream&statsPeriod=14d&stream_index=0) where the sync script succeeds in creating the initial Product entry the first time through, but fails to update existing Product on subsequent runs. That particular issue with Product upsert was fixed, but there is a similar problem with the way we're upserting Deals. 

## Proposed Changes

The usage of Django's `update_or_create` was incorrect for upserting Deals, and this PR updates to the correct usage.
I also separated out the mocks into their own files outside the test file, to make it a bit more manageable to work with the tests.

## Verification Steps

These specs verify the new usage, but they don't explicitly verify that when the script encounters an existing deal, the upsert behavior is correct. To do this, I'd want to refactor the `upsert_deals` function into several pieces so it'd be easier to test this kind of behavior, but I held off on that in this PR. 



[SHEP-37]: https://mozilla-hub.atlassian.net/browse/SHEP-37?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
